### PR TITLE
fix: inconsistent image normalization for wandb.Image

### LINKED ISSUE #6287: [CLI]: wandb.Image gives blanks for large or small values
### Describe the bug

According to the [wandb.Image documentation](https://docs.wandb.ai/ref/python/data-types/image), the type accepts "numpy array of image data", where it "attempts to infer the data format and converts it". This works for float-valued raw data, but not if the values are very large or very small in scale. [The code doing the conversion](https://github.com/wandb/wandb/blob/main/wandb/sdk/data_types/image.py#L269) has different pathways for numpy arrays and tensors. Torch tensors work properly for large values, but not for small values.

If an array of values of scale 1e-5 shows as white-to-black noise, it doesn't make much sense that an array of 1e-8 values would show as full-black.

```python
>>> import numpy as np
>>> import wandb
>>> wandb.Image(np.random.uniform(size=[100, 200]) * 1e-8).image.save("preview_e-8.png")
>>> wandb.Image(np.random.uniform(size=[100, 200])).image.save("preview_e0.png")
>>> wandb.Image(np.random.uniform(size=[100, 200]) * 1e8).image.save("preview_e8.png")
>>> wandb.Image(torch.rand(size=[100, 200]) * 1e-8).image.save("torch_preview_e-8.png")
>>> wandb.Image(torch.rand(size=[100, 200])* 1e-5).image.save("preview_e-5.png")
>>> wandb.Image(torch.rand(size=[100, 200]) * 1e8).image.save("torch_preview_e8.png")
```

No exception, but the images look very different even though they should look qualitatively similar. `preview_e-8.png` is a fully black image, `preview_e0.png` is as expected (noise), and `preview_e8.png` is a fully white image. `preview_e-5.png` is the first scale where we get noise, at e-6 we don't get black but the white values become dark gray. The conversion works for large torch tensors, and `torch_preview_e8.png` shows noise, but `torch_preview_e-8.png` is a black image.

### Additional Files

_No response_

### Environment

WandB version: 0.15.0

OS: macOS 13.0 Ventura

Python version: 3.10.10

Versions of relevant libraries:

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by jacobromero**
Description
-----------

- Fixes WB-15801
- Fixes #6287, #7377

What does the PR do? Include a concise description of the PR contents.
When normalizing image values, during the creation of a `wandb.Image` from data (numpy array, or pytorch tensor) there are inconsistencies in the output image. 

Specifically tensorflow will always do a min/max normalization. While for numpy arrays we would only do a min/max normalization when the array contained a negative value. This led to some images being completely dark for numpy data, but normalized for tensorflow data. This PR aligns the behavior for both numpy and tensorflow, and adds a new `normalize` parameter to prevent W&B from modifying the data at all.

- [ ] I updated CHANGELOG.unreleased.md, or it's not applicable

Testing
-------
How was this PR tested?
- Unit tests added
- repro script
```python
import numpy as np
import wandb
import torch

img = np.random.uniform(low=0, high=1, size=[2, 4, 3])
torch_img = torch.from_numpy(img.transpose(2, 0, 1))
values = [1e-8, 1e-5, 1e0, 1e1]

with wandb.init(
    project="image-norm-test",
) as run:
    for value in values:
        wandb_np_img_norm = wandb.Image(img * value)
        wandb_torch_img_norm = wandb.Image(torch_img * value)

        run.log({f"np_img_norm_{str(value)}": wandb_np_img_norm})
        run.log({f"torch_img_norm_{str(value)}": wandb_torch_img_norm})
```

#### After this change
image
image
image
image

#### Before this change
image
image
image
image

### CODE REVIEW

File: wandb/sdk/data_types/image.py [hunk 1]
```diff
@@ -508,29 +560,30 @@ def guess_mode(
             )
 
     @classmethod
-    def to_uint8(cls, data: "np.ndarray") -> "np.ndarray":
-        """Convert image data to uint8.
-
-        Convert floating point image on the range [0,1] and integer images on the range
-        [0,255] to uint8, clipping if necessary.
-        """
+    def normalize(cls, data: "np.ndarray") -> "np.ndarray":
```
Comment (timoffex) [inline wandb/sdk/data_types/image.py:563]: Should be prefixed by an underscore since it's private, and can be a top-level function rather than a classmethod

File: wandb/sdk/data_types/image.py [hunk 2]
```diff
@@ -508,29 +560,30 @@ def guess_mode(
...
         if np.max(data) <= 1.0:
-            data = (data * 255).astype(np.int32)
+            data = (255 * data).astype(np.int32)
 
-        # assert issubclass(data.dtype.type, np.integer), 'Illegal image format.'
-        return data.clip(0, 255).astype(np.uint8)
+        return data.clip(0, 255)
+
+    @classmethod
+    def convert_to_uint8(cls, data: "np.ndarray") -> "np.ndarray":
```
Comment (timoffex) [inline wandb/sdk/data_types/image.py:581]: Likewise, should be prefixed with an underscore

Also, doesn't need to be a classmethod

**Review by timoffex — APPROVED** [review]
